hack/generator/pkg/codegen/code_generator.go

```diff
@@ -56,6 +56,7 @@ func NewCodeGeneratorFromConfig(configuration *config.Configuration, idFactory a
 func corePipelineStages(idFactory astmodel.IdentifierFactory, configuration *config.Configuration) []PipelineStage {
 	return []PipelineStage{
 		nameTypesForCRD(idFactory),
+		improveResourcePluralization(),
```

Comment:
will take suggestions for a better name =)

Reply:
How about `singularizeResourceNames`? (Or even better `singularise` although that's probably a bridge too far. ;)
Then my fix would include to changing it to `maybeSingularizeResourceNames`, which is quite a mouthful.

I quite like the vagueness of `improve` though.